src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -97,19 +97,36 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 		
 		BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
 		if (type != null) {
+			//Gets all modules of the type specified by variable type
```

Comment:
As a general style guide, you should never have a code comment that does not provide extra information to the reader.
This comment is bad for two reasons:

1. don't say "by variable type" because it's very obvious to someone looking at the code that type is a variable, and because of its name it's self-documenting
2. the whole line doesn't provide any additional info that you couldn't get from looking at the one line of code.

The thing that _would_ be useful to say in this comment is to provide a bit of information to someone who may be reading this, but is unfamiliar with the ES query API. E.g. most people won't know specifically what a "match" query is. So I would have written as the comment here:

    // exact match on type

This would provide the reader with the most information about what this does, while taking the least time to read.

Also, as I type this, I don't see why we didn't use a `term` query here instead of a `match`.